src/SwiftPackage.ts

```diff
@@ -45,39 +45,55 @@ export interface Dependency {
     url?: string
 }
 
+// package we attempted to load but failed
+class InvalidPackage implements PackageContents {
+    get name(): string { return ""; }
+    get products(): Product[] { return []; }
+    get dependencies(): Dependency[] { return []; }
+    get targets(): Target[] { return []; }
+}
+
 // Class holding Swift Package Manager Package
 export class SwiftPackage implements PackageContents {
 	private constructor(
         readonly folder: string,
-        public contents?: PackageContents
+        public contents?: PackageContents|null
     ) {
         this.setContextKeys();
     }
 
     public static async create(folder: string): Promise<SwiftPackage> {
+        let contents = await SwiftPackage.loadPackage(folder);
+        return new SwiftPackage(folder, undefined);
```

Comment:
`undefined` should be `contents` here I think?

Reply:
Good catch, cut and paste coding :-(